linkatos.py

```diff
@@ -38,67 +38,33 @@ def store_link(link, channel):
     return None
 
 
-def parse_output(slack_rtm_output):
-    """
-        The Slack Real Time Messaging API is an events firehose.
-        this parsing function returns None unless
-        someone posts a website link starting with httpS?// or a yes
-    """
-    # default outcome
-    finding = None
-    channel = None
-    output_type = None
-
-    output_list = slack_rtm_output
-    print(output_list)  # print the list of outputs to get them on screen
-
-    if output_list and len(output_list) > 0:
-        for output in output_list:
-            # when there is a message then get the channel
-            if output and 'text' in output and output['user'] != BOT_ID:
-                finding = message.extract_url(output['text'])
-                if finding is not None:
-                    output_type = 'link'
-                else:
-                    if message.has_a_yes(output['text']):
-                        output_type = 'yes'
-
-            if output and 'channel' in output:
-                channel = output['channel']
-
-    return (finding, channel, output_type)
-
-
 if __name__ == '__main__':
     READ_WEBSOCKET_DELAY = 1  # 1 second delay between reading from firehose
 
     # verify linkatos connection
     if slack_client.rtm_connect():
         print("linkatos is connected and running!")
+        expecting_confirmation = False
 
         while True:
             print("linkatos is listening")
 
             # parse the messages. Get 'None' while they're empty
-            messages = slack_client.rtm_read()
-            (link, channel, output_type) = parse_output(messages)
-
-            # handle the command when it is a http address
-            if link is not None and output_type is 'link' and channel:
-                bot_says(channel, "Do you want me to store the link " +
-                         link + " for you?")
-
-                # parse answerif answer...
-                messages = slack_client.rtm_read()
-                (answer, channel, output_type) = parse_output(messages)
-                print(answer)
-
-                # just to debug
-                if answer is not None:
-                    print("you said Yes")
-
-                # if response:
-                #     store_link(link)
+            (out, channel, out_type) = message.parse(slack_client.rtm_read())
+
+            # handle the command either expecting a url or a yn_answer
+            if out is not None and channel:
+                if expecting_confirmation is False and out_type is 'url':
```

Comment:
What happens if `expecting_confirmation is False and out_type is 'yn_answer'`?

If it must not happen, we should try to model the data so there is no room for that.